TransformationSystem/DB/TransformationDB.py

```diff
@@ -173,9 +176,27 @@ def addTransformation(self, transName, description, longDescription, authorDN, a
       return res
     transID = res['lastRowId']
     self.lock.release()
+
+    # Add Input and Output Meta Queries to the transformation ###############
+    if inputMetaQuery is None:
+      inputMetaQuery = {}
+    if outputMetaQuery is None:
+      outputMetaQuery = {}
+
+    if len(inputMetaQuery) > 0:
+      res = self.createTransformationMetaQuery(transID, inputMetaQuery, 'Input')
+      if not res['OK']:
+        gLogger.error("Failed to add input meta query to the transformation", res['Message'])
+        return self.deleteTransformation(transID, connection=connection)
+    if len(outputMetaQuery) > 0:
+      res = self.createTransformationMetaQuery(transID, outputMetaQuery, 'Output')
+      if not res['OK']:
+        gLogger.error("Failed to add output meta query to the transformation", res['Message'])
+        return self.deleteTransformation(transID, connection=connection)
+
     # If the transformation has an input data specification
-    if fileMask:
-      self.filters.append((transID, json.loads(fileMask)))
+    if len(inputMetaQuery) > 0:
```

Comment:
No need for `len`

Reply:
Done:
8b0ea690e093080f21879899ec80301b0326996e